WcaOnRails/app/models/user.rb

```diff
@@ -861,6 +861,20 @@ def notify_of_results_posted(competition)
     end
   end
 
+  def maybe_assign_wca_id_by_results(competition, notify = true)
+    if !wca_id && !unconfirmed_wca_id
+      matches = []
+      unless country.nil?
+        matches = competition.competitors.where(name: name, year: dob.year, month: dob.month, day: dob.day, gender: gender, countryId: country.id).to_a
```

Comment:
As far as I understand, @SAuroux's concern is that we us `competition.competitors` here, which is reasonable when posting results of a single competition, but if doing a massive assignment for each competition, then there may be two competitions with different accounts matching the same person profile. For the massive assignment it would make more sense to search all the people instead.

Reply:
@SAuroux @jonatanklosko hmm I see, you guys are right. Then in the rails console command in the first post that goes over all the competitions we'll need to order by the competition date descending.

Reply:
Sounds good to me. Also, I think it would be best to make this a migration for the reference.

Reply:
Sounds good to me too!

Reply:
Updated the code in the first post with the ordering.

Following a slack chat with Jonatan about using a rails migration, I looked into it further, and a migration would block page loads until it's complete, which could be painful. There are alternative solutions out there like using an async rails migrations gem, or create one-time rake tasks. Could be a bit of an overkill, though.

It's up to you guys how to go about it, let me know :)

Reply:
Yeah, I agree these solutions would be an overkill, happy to hear @viroulep thoughts though!

Reply:
I'm absolutely fine with running this in the rails console while deploying the changes!
The rails migration would not be reversible anyway.
We definitely want to trigger a backup of the db right before doing that though (we need to find the appropriate command in the cronjob).
According to the various recipes it should be something along the line of:
```
scripts/db.sh dump secrets/wca_db && scripts/backup.sh
```